lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb

```diff
@@ -0,0 +1,139 @@
+require 'fileutils'
+
+Puppet::Type.type(:bootstrap_rpm).provide(:bootstrap_rpm) do
+
+  commands :rpmbuild => 'rpmbuild'
+  commands :rpm => 'rpm'
+
+  def create
+    copy_sources
+    write_specfile
+    build_rpm
+    copy_rpm
+    link_rpm
+  ensure
+    FileUtils.remove_dir(base_dir) if File.directory?(base_dir)
+    @base_dir = nil
+  end
+
+  def delete
+  end
+
+  def exists?
+    !resource[:symlink] || File.exist?(resource[:symlink])
+  end
+
+  private
+
+  def base_dir
+    @base_dir ||= Dir.mktmpdir
+  end
+
+  def sources_dir
+    mkdir('SOURCES')
+  end
+
+  def spec_dir
+    mkdir('SPECS')
+  end
+
+  def rpm_dir
+    mkdir('RPMS')
+  end
+
+  def copy_sources
+    FileUtils.copy(resource[:script], File.join(sources_dir, File.basename(resource[:script])))
+  end
+
+  def write_specfile
+    File.open(File.join(spec_dir, "#{resource[:name]}.spec"), 'w') do |file|
+      file.write(spec)
+    end
+  end
+
+  def release
+    if !latest_rpm.nil? && File.exist?(latest_rpm)
+      rel = rpm('-qp', latest_rpm, "--queryformat=%{release}")
+      rel.to_i + 1
+    else
+      1
+    end
+  end
+
+  def build_rpm
+    output = rpmbuild(
+      '-ba',
+      File.join(spec_dir, "#{resource[:name]}.spec"),
+      '--define', "_topdir #{base_dir}",
+      '--define', "name #{resource[:name]}",
+      '--define', "release #{release}"
+    )
+
+    Puppet.debug(output)
+  end
+
+  def copy_rpm
+    Dir.glob("#{rpm_dir}/noarch/*.rpm") do |file|
+      FileUtils.copy(file, resource[:dest])
+    end
+  end
+
+  def link_rpm
+    FileUtils.ln_s(latest_rpm, resource[:symlink], force: true) if resource[:symlink]
+  end
+
+  def latest_rpm
+    latest = Dir.glob("#{resource[:dest]}/*.rpm").sort.reverse
+    false if latest.nil?
```

Comment:
Did you intend to return?
```suggestion
    return false if latest.nil?
```